### PR TITLE
feat: add `Order` on effect `Atoms`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
This is part A of the fix in #9021

File: main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala [hunk 1]
```diff
@@ -290,15 +311,18 @@ object EffUnification3 {
...
+    def getAtoms(t: Type)(implicit scope: Scope, renv: RigidityEnv): SortedSet[Atom] = t match {
+      case Type.Var(sym, _) if renv.isRigid(sym) => SortedSet(Atom.VarRigid(sym))
+      case Type.Var(sym, _) => SortedSet(Atom.VarFlex(sym))
+      case Type.Cst(TypeConstructor.Effect(sym), _) => SortedSet(Atom.Eff(sym))
+      case Type.Cst(TypeConstructor.Error(id, _), _) => SortedSet(Atom.Error(id))
       case Type.Apply(tpe1, tpe2, _) => getAtoms(tpe1) ++ getAtoms(tpe2)
       case Type.Alias(_, _, tpe, _) => getAtoms(tpe)
-      case assoc@Type.AssocType(_, _, _, _) => getAssocAtoms(assoc).toSet
-      case _ => Set.empty
+      case assoc@Type.AssocType(_, _, _, _) => getAssocAtoms(assoc) match {
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala:321]: Is there no `to(SortedSet)`?

File: main/src/ca/uwaterloo/flix/language/phase/unification/set/SetUnification.scala [hunk 2]
```diff
@@ -536,7 +536,7 @@ object SetUnification {
 
 
     val query = mkEmptyQuery(eq.f1, eq.f2)
-    val fvs = query.variables.toList
+    val fvs = query.variables.toList.reverse
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/unification/set/SetUnification.scala:539]: This could do with a comment.

**Review by mlutze — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala (hunk 1 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala:321]: It could be `SortedSet.from(getAssocAtoms(assoc))` im pretty sure

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala:321]: PR welcome

**Pull request merged**